### DISCUSSION TITLE
Sharing successful Hauptwerk sample sets usage in GrandOrgue

### DISCUSSION THREAD

**Discussion by eturpault** [GrandOrgue/OdfEdit#16] [Show and tell]
Discussion created if people want to share successful conversions from Hauptwerk ODF to GrandOrgue ODF by OdfEdit tool, and the obtained result in GrandOrgue.
It's nice to also share the success, not just the problems :smiley:.

**Comment by rousseldenis** [comment]
@eturpault I nearly want to buy Dom Bedos sample set from sonusparadisi. Do you know if the current version of OdfEdit could convert it?

**Comment by eturpault** [comment]
Yes current version of OdfEdit should convert it without issue, I tested it successfully (with the HW ODF alone, without the package files) after that some maximum attribute values have been increased in GO SW on my request if I remember well.
You should have the same result as the one reported here : https://github.com/GrandOrgue/OdfEdit/discussions/2#discussioncomment-4869950
Anyway if you encounter an issue please report it to me and I will fix it.
In my side I bought the Prytanée and Frankfurt sample sets from Sonus Paradisi and thanks to OdfEdit it is a pleasure to play them with GO, they are very nice organs with high quality samples recording. I have not converted the tremmed samples because it consume too much RAM in the computer running GO.

**Comment by larspalo** [comment]
I've bought the full St. Michel en Thiérache from sonusparadisi.cz and can confirm that OdfEdit 2.4 does convert it into a fully playable organ for GrandOrgue. There are a few small issues but nothing really serious and they can relatively easily be fixed. In the odf I notice a few things that could be done in a better way, but they are of less importance than the fact that the tool does it's job! Thanks!

**Reply by eturpault** [reply]
Don't hesitate to let me know what OdfEdit could do in a better way when building the GO ODF (maybe it is what you already did in two other comments today), I implemented it with my own experience of the GO ODF usage which is not perfect. I will evaluate what can be improved. Thanks.

**Comment by davidgritter** [comment]
I have converted the Sonus Paradisi skinner opus 497, and got a more or less working odf file.  I note the following problems, however:

chimes, harp, celeste ranks do not get converted with the percussive attibute set to true

This sample set has both landscape and portrait stop layout panels.  both landscape and portrait panels have the same name.  Only the landscape panels were imported into the odf.

The sample set has 4 enclosures tied to swell shoes.    The odf is structured with only one swell enclosure, mapped to the three divisions that are enclosed.

In this sample set the echo division has no keyboard manual and no couplers.  Normally it is considered part of the solo division, but it is mapped as its own division in the odf with no way to play it.

There is some complicated logic in the original xml file where switches on the mixer panel control which rank is played from two of the switches in the other panels (fife1 and gt mixture).  This logic is mangled in translation so that both ranks play simultaneously.

None of these problems appears difficult to fix using GOODF, except for the missing panels.

I also find it a mystery that the converted  odf has approximately the right volume levels for all the stops.  If I create an ODF manually using tools like GOODF, or jmakeODF some of the stops( for example the solo Gamba) needs to have their gain boosted by over 12dB in order to sound properly.  I have scanned the converted ODF, and find gain adjustments on some of the individual pipes, but no overall gain boost such as I needed manually.  Does anyone know where such a gain boost might be coming from in the ODFEdit conversion process or where I might find it in the ODF?

**Reply by eturpault** [reply]
This sample set seems to be a good example of additional features not supported by OdfEdit HW to GO conversion so far.
As written in the readme of this project, I think that it will never be possible to convert for GO a sample set with all the features which he has in HW, expecially the complex controls, but I will study what can be added to support more features of this sample set.
About the gain of the pipes, OdfEdit sets it in the GO ODF from the attribute AmpLvl_LevelAdjustDecibels of each corresponding pipe of the HW ODF, if it is higher than 0.
About the alternate screen layouts (panels in portrait mode usually), I don't know if it is worth to convert them in the GO ODF, for the moment I confirm that only the default layouts (landscape mode usually) are converted.

**Reply by eturpault** [reply]
The alternate layout panels will be converted from HW to GO ODF if enabled in a new menu option of OdfEdit v2.8.

**Comment by pgwid-og** [comment]
I succesfully converted the SP Santanyi (https://www.sonusparadisi.cz/en/organs/spain/santanyi.html) or to be more precise an extension (https://www.sonusparadisi.cz/en/blog/santanyi-portrait-screens-available/) of this old but nice Spanish sample set.
Now I will be able (what never happened in HW) to add a simple screen (the original layout is a bit confusing, even if one used a longer time). *** smile ***

**Comment by Psi58** [comment]
I've successfully converted the demo version of Ivan Barritt's St David's Cathedral organ (https://barrittaudio.co.uk/pages/st-davids-cathedral) - ODFEdit did a great job, with just one path that needed to be manually corrected.
So now I'm contemplating buying the full version, before the price goes up on Saturday.
Has anyone tried this, who can reassure me that it will work? It would be quite an expensive mistake if the conversion fails...
Thanks!

**Reply by eturpault** [reply]
Thanks for OdfEdit 😃.
Personnaly I have bought the sample sets Prytanée and Frankfurt of Sonus Paradisi after having tried their demo version. The full versions are designed like the demo versions and with several stops more. Except the stops number there is no other differences, and the full versions are working very well.
So if the convertion from HW to GO has worked well with the St David's Cathedral demo version, I don't expect problems with its full version. If you have convertion issues with the full version you can provide me only the ODF of this version with the description of the issue and I will try to fix the issue in OdfEdit.

I have just downloaded and tested the demo version of St David's Cathedral. The loading in GO generates a lot of errors or warnings "temperament would retune pipe by" or "no pitch information provided" when the option "Perform strict ODF" is checked in the settings of GO, but this is not preventing the sample set to work well in GO. It may cause problems when changing the temperament or the pitch of the organ in GO.
The path issue is related to the noise folders which have a semi-colon character inside their name in the sample set, like "Noises; Blower", "Noises; ECh. Tremulant", ... It is very strange to use a semi-colon in a folder name, and the first time I see that.
The semi-colon is used in the GO ODF syntax to mark the beginning of a comment, so it is causing misbehaviors when loading the sample set in GO, the noise sample files are not found by GO.
So all semi-colons characters present in rank sections must be removed in the GO ODF and have to be removed as well in the corresponding folders name of the sample set. This has to be done manually, my rule is that OdfEdit does not modify the original sample set when doing the convertion to GO ODF.
Once this manual modification is done, the demo sample set and its noises are working very well.

What I see in this demo sample set is that all the stops of the full version are defined in the demo ODF, but the wav files of the stops not available in the demo version have a length of 0, so they are not emitting sound. You can see for example in the folder OrganInstallationPackages\002933\Samples\Gt. Double Diapason 16\Front, all .wav files have a size of 145 bytes only.
The full version should work without issue I think, it seems that there are no ODF design differences between demo and full versions, only the .wav files content are different.

**Reply by Psi58** [reply]
Many thanks @eturpault, that's very helpful.
It's unlikely I would want to change the temperament or the pitch, so that's not an issue for me.
And the path issue I resolved exactly as you described - relocating the noise subfolders, and replacing the semi-colons in the paths.
I will post another update if there are any other issues - but I hope that it will all be OK...
Thanks again for OdfEdit!

**Reply by Psi58** [reply]
I'm not the only one working on this - see https://mps-orgelseite.de/home/filebase/index.php?file/276-go-odf-for-st-davids-cathedral-barritt-audio-full-version/#overview. Also using OdfEdit 😉

**Reply by eturpault** [reply]
It's good to know that OdfEdit is usefull for people, thanks ! 
I see in the forum of mps-orgelseite.de that OdfEdit is mentioned several times.
I am avaible to improve it if needed.

**Comment by BasKlb** [comment]
- I tried it with te demo of the Pinerolo sampleset by Piotr Gabrowski. The resulting ODF wasn't perfect (yet). For example, only the direct perspective was loaded, although the front and rear samples were also available. This organ has a split in the manual that was not effectuated. So some editing was needed, for which I used Notepad++ . The result has been uploaded to the Filebase of MPS Orgelseite.

-  Last week I tried to generate a GO-Odf for the Prospectum sampleset of the Frauenkirche. This however was unsuccesfull.  The HW Odf is not an .XML file but has the extension:" .Organ_Hauptwerk_hbx".

**Reply by eturpault** [reply]
In the demo HW ODF, only ranks of "close" samples are linked to the stops, OdfEdit can convert in the GO ODF only what is described in the HW ODF. But as ranks of "front" and "rear" samples are defined in the HW ODF (but are not linked to stops), you can load the HW ODF after having enabled the option "convert unused ranks" of OdfEdit, then move manually by drag&drop these ranks under their respective stop in OdfEdit.

Could you elaborate more about the split which is defined for this organ ? Is it at some stops level, in which manual ?

Is the ODF with extension ".Organ_Hauptwerk_hbx" a regular HW ODF xml file ? If yes, I can add the support of this extension in OdfEdit.

**Reply by eturpault** [reply]
Has someone on this forum the Hauptwerk application and has tried to use the Pinerolo sampleset by Piotr Grabowski ? Are the front and rear samples actually played by Hauptwerk ? I analyzed again the HW ODF of this sampleset, I don't see how front and rear ranks could be linked to their corresponding stop. This ODF has none StopRank section defined, which is used to link ranks to stops in all samplesets (but last demo samplesets of Piotr Grabowski).

**Comment by medical990** [comment]
I would like to share my successes here.

Success
- Jeux d'orgues 3
- Jeux d'orgues 2 (Stiehr-mockers, Free)
- Azzio (PG, Free)
- Strassburg (PG, Free)
- Lorris (AVO organ; Free)
- Lemmer, Flentrop (AVO organ; Free)
- Zoeblitz Silbermann HW version (Prospectum)
- Smecno DEMO (SP)
- Arnstadt DEMO (SP)
- Menesterol DEMO (SP; It is actually a full stereo set, only lacking in surround function).
- Galician Organ (SP; Free)
- Kdousov Shareware (SP; It is actually a free full set. The message asking for donation appears, without limiting the functions.)
- Prague Baroque DEMO

**Reply by eturpault** [reply]
It's nice to know that OdfEdit is being used successfully, thank you for this feedback.
